temporal/api/cloud/nexus/v1/message.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+
+package temporal.api.cloud.nexus.v1;
+
+option go_package = "go.temporal.io/api/cloud/nexus/v1;nexus";
+
+import "google/protobuf/timestamp.proto";
+
+message IncomingServiceSpec {
+    // The name of the incoming service. Must be unique within an account
+    // The name must match `[a-zA-Z_][a-zA-Z0-9_]*`.
+    string name = 1;
+
+    // The namespace to route requests to
+    string namespace = 2;
+
+    // The task queue to route requests to
+    string task_queue = 3;
+
+    // The authorization specification for this service
+    IncomingServiceAuthorizationSpec authorization_spec = 4;
+}
+
+message IncomingServiceAuthorizationSpec {
+    AllowedNamespacePolicySpec allowed_namespace_policy_spec = 1;
+}
+
+message AllowedNamespacePolicySpec {
+    // The set of namespaces that are allowed to call into this service
+    repeated string namespaces = 1;
+}
+
+// A binding from a service name to namespace and task queue for dispatching incoming Nexus requests.
+message IncomingService {
+    // The id of the service. This is generated by the server and is immutable
+    string id = 1;
+
+    // The current version of the incoming service specification
+    // The next update operation must include this version
+    string resource_version = 2;
+
+    // The incoming service specification
+    IncomingServiceSpec spec = 3;
+
+    // The current state of the service
+    string state = 4;
+
+    // The id of the async operation that is creating/updating/deleting the service, if any
+    string async_operation_id = 5;
+
+    // The date and time when the service was created
+    google.protobuf.Timestamp created_time = 6;
+
+    // The date and time when the service was last modified
+    // Will not be set if the service has never been modified.
+    google.protobuf.Timestamp last_modified_time = 7;
+}
+
+message OutgoingServiceSpec {
+    // The name of the outgoing service. Must be unique within the namespace
+    // The name must match `[a-zA-Z_][a-zA-Z0-9_]*`.
+    // This name is immutable
+    string name = 1;
+
+    BindingSpec spec = 2;
+}
+
+message BindingSpec {
+    IntraAccountServiceSpec intra_account_service_spec = 1;
+}
+
+message IntraAccountServiceSpec {
```

Comment:
in the future we could add GenericURI spec or CrossAccountServiceSpec of w/e else?

Reply:
IntraCloudServiceSpec ?